cclib/parser/fchkparser.py

```diff
@@ -0,0 +1,239 @@
+            count = int(line.split()[-1])
+
+            # triangle matrix, with number of elements in a row:
+            # 1 + 2 + 3 + .... + self.nbasis
+            assert count == (self.nbasis + 1) * self.nbasis / 2
+            raw_overlaps = self._parse_block(inputfile, count, float, 'Overlap Matrix')
+
+            # now turn into matrix
+            overlaps = numpy.zeros((self.nbasis, self.nbasis))
+            raw_index = 0
+            for row in range(self.nbasis):
+                for col in range(row + 1):
+                    overlaps[row, col] = raw_overlaps[raw_index]
+                    overlaps[col, row] = raw_overlaps[raw_index]
+                    raw_index += 1
+
+            print(overlaps[0:5, 0:5])
+            self.set_attribute('aooverlaps', overlaps)
+
+        if line[0:31] == 'Number of independent functions':
+            self.set_attribute('nmo', int(line.split()[-1]))
+
+        if line[0:21] == 'Alpha MO coefficients':
+            count = int(line.split()[-1])
+            assert count == self.nbasis * self.nmo
+
+            coeffs = numpy.array(self._parse_block(inputfile, count, float, 'Alpha Coefficients'))
+            coeffs.shape = (self.nmo, self.nbasis)
+            self.set_attribute('mocoeffs', [coeffs])
+
+        if line[0:22] == 'Alpha Orbital Energies':
+            count = int(line.split()[-1])
+            assert count == self.nmo
+
+            energies = numpy.array(self._parse_block(inputfile, count, float, 'Alpha MO Energies'))
+            self.set_attribute('moenergies', [energies])
+
+        if line[0:20] == 'Beta MO coefficients':
+            count = int(line.split()[-1])
+            assert count == self.nbasis * self.nmo
+
+            coeffs = numpy.array(self._parse_block(inputfile, count, float, 'Beta Coefficients'))
+            coeffs.shape = (self.nmo, self.nbasis)
+            self.append_attribute('mocoeffs', coeffs)
+
+        if line[0:21] == 'Beta Orbital Energies':
+            count = int(line.split()[-1])
+            assert count == self.nmo
+
+            energies = numpy.array(self._parse_block(inputfile, count, float, 'Alpha MO Energies'))
+            self.append_attribute('moenergies', energies)
+
+        if line[0:11] == 'Shell types':
+            self.parse_aonames(line, inputfile)
+
+    def parse_aonames(self, line, inputfile):
+        # e.g.: Shell types                                I   N=          28
+        count = int(line.split()[-1])
+        shell_types = self._parse_block(inputfile, count, int, 'Atomic Orbital Names')
+
+        # e.g.: Number of primitives per shell             I   N=          28
+        next(inputfile)
+        self._parse_block(inputfile, count, int, 'Atomic Orbital Names')
+
+        # e.g. Shell to atom map                          I   N=          28
+        next(inputfile)
+        shell_map = self._parse_block(inputfile, count, int, 'Atomic Orbital Names')
+        table = utils.PeriodicTable()
```

Comment:
This is already available as `self.table`: https://github.com/cclib/cclib/blob/f1aaf2788a7903f732887a1c5a0131c4759ae6eb/cclib/parser/logfileparser.py#L237